src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DocumentState.cs

```diff
@@ -67,42 +69,35 @@ internal class DocumentState
 
         public GeneratedCodeContainer GeneratedCodeContainer => HostDocument.GeneratedCodeContainer;
 
-        public DocumentGeneratedOutputTracker GeneratedOutput
+        public bool IsGeneratedOutputResultAvailable => ComputedState.TaskUnsafe?.IsCompleted == true;
+
+        private ComputedStateTracker ComputedState
         {
             get
             {
-                if (_generatedOutput == null)
+                if (_computedState == null)
                 {
                     lock (_lock)
                     {
-                        if (_generatedOutput == null)
+                        if (_computedState == null)
                         {
-                            _generatedOutput = new DocumentGeneratedOutputTracker(null);
+                            _computedState = new ComputedStateTracker(this);
                         }
                     }
                 }
 
-                return _generatedOutput;
+                return _computedState;
             }
         }
 
-        public DocumentImportsTracker Imports
+        public Task<(RazorCodeDocument output, VersionStamp inputVersion, VersionStamp outputVersion)> GetGeneratedOutputAndVersionAsync(DefaultProjectSnapshot project, DefaultDocumentSnapshot document)
```

Comment:
Can the second parameter just be `DocumentSnapshot` instead of `DefaultDocumentSnapshot`?

Reply:
I'm sure it could be, it should not be. This code is part of **the implementation**, it should not be reusable.